src/API/QueryFieldService.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace EzSystems\EzPlatformQueryFieldType\API;
+
+use eZ\Publish\API\Repository\ContentTypeService;
+use eZ\Publish\API\Repository\SearchService;
+use eZ\Publish\API\Repository\Values\Content\Content;
+use eZ\Publish\API\Repository\Values\Content\ContentInfo;
+use eZ\Publish\API\Repository\Values\Content\Search\SearchHit;
+use eZ\Publish\API\Repository\Values\ContentType\FieldDefinition;
+use eZ\Publish\Core\QueryType\QueryTypeRegistry;
+use Symfony\Component\ExpressionLanguage\ExpressionLanguage;
+use Symfony\Component\Yaml\Yaml;
+
+/**
+ * Executes a query and returns the results.
+ */
+class QueryFieldService
+{
+    /** @var \eZ\Publish\Core\QueryType\QueryTypeRegistry */
+    private $queryTypeRegistry;
+
+    /** @var \eZ\Publish\API\Repository\SearchService */
+    private $searchService;
+
+    /** @var \eZ\Publish\API\Repository\ContentTypeService */
+    private $contentTypeService;
+
+    public function __construct(
+        SearchService $searchService,
+        ContentTypeService $contentTypeService,
+        QueryTypeRegistry $queryTypeRegistry
+    ) {
+        $this->searchService = $searchService;
+        $this->contentTypeService = $contentTypeService;
+        $this->queryTypeRegistry = $queryTypeRegistry;
+    }
+
+    /**
+     * @param \eZ\Publish\API\Repository\Values\Content\Content $content
+     * @param string $fieldDefinitionIdentifier
+     *
+     * @return \eZ\Publish\API\Repository\Values\Content\Content[]
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException
+     */
+    public function loadFieldData(Content $content, string $fieldDefinitionIdentifier): array
```

Comment:
```suggestion
    public function loadFieldData(Content $content, string $fieldDefinitionIdentifier): iterable
```

_To make possible to return collection object in the future_